app/frontend/src/service/resources.ts

```diff
@@ -6,3 +6,23 @@ export async function getTermsOfService() {
 
   return res.toObject();
 }
+
+export async function getRegions() {
+  return client.resources.getRegions(new Empty()).then(
+    (res) => res.toObject(),
+    (error) => {
+      console.error("Failed to get regions.");
+      throw error;
+    }
+  );
+}
+
+export async function getLanguages() {
+  return client.resources.getLanguages(new Empty()).then(
+    (res) => res.toObject(),
+    (error) => {
+      console.error("Failed to get languages.");
+      throw error;
+    }
+  );
```

Comment:
You can do await on `getRegions` and `getLanguages`, then return `res.toObject()`. No need to catch the error and re-throw it as React Query surfaces that in the `error` property returned from `useQuery`. 

Also remove `console.error`, it just clutters the console on top of errors we'll get from the ones React Query forwards from grpc to the console so I'd remove it.